### PR TITLE
adding guest login #646 - disable not ready yet

### PR DESCRIPTION
**Pull request by jdelrue**
Standard guest login, todo disable with config!

### CODE REVIEW

File: src/views/Login.vue [hunk 1]
```diff
@@ -19,15 +26,19 @@ export default {
...
   methods: {
     ...mapActions([
       'generateLoginUrl',
       'checkResponse'
-    ])
+    ]),
+    threebotConnectLogin: function() {
+      this.generateLoginUrl(this.$route.query)
+    },
+    guestLogin: function() {
```
Comment (JonasWijne) [inline src/views/Login.vue:39]: simplify to `guestLogin() {`

File: src/views/Login.vue [hunk 2]
```diff
@@ -19,15 +26,19 @@ export default {
...
+      this.$router.push( { path: "guestlogin", query: { 'redirect' : this.$route.query.redirect}});
```
Comment (JonasWijne) [inline src/views/Login.vue:40]: add some newlines

File: src/views/Login.vue [hunk 3]
```diff
@@ -1,15 +1,22 @@
...
         </v-card-text>
-        <v-card-text v-else>
-          Redirecting to login...
-        </v-card-text>
+        <v-container  v-else>
+          <v-card-text>
+            Please login using 3Bot Connect or continue as guest.
+          </v-card-text>
+          <v-card-actions>
+            <v-btn v-on:click="threebotConnectLogin" text>Use 3Bot Connect</v-btn>
```
Comment (JonasWijne) [inline src/views/Login.vue:15]: change `v-on:click` to `@click`

File: src/views/Login.vue [hunk 4]
```diff
@@ -1,15 +1,22 @@
...
+            <v-spacer></v-spacer>
+            <v-btn v-on:click="guestLogin" text>Continue as Guest</v-btn>
```
Comment (JonasWijne) [inline src/views/Login.vue:17]: change `v-on:click` to `@click`

File: src/views/Login.vue [hunk 5]
```diff
@@ -1,15 +1,22 @@
 <template>
   <v-row align="center" justify="center">
-      <v-card loading width="500">
+      <v-card :loading="$route.query.callback" width="500">
         <v-card-title>
-          Login
+          Freeflow Connect
```
Comment (JonasWijne) [inline src/views/Login.vue:5]: should the filename / routename be changed ?

File: src/views/GuestLogin.vue [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+        this.$router.push(this.$route.query.redirect);
+      } else {
+        //   this.$router.push({ name: "room" });
+      }
+    }
+  },
+  computed: {
+    ...mapGetters(["loginUrl", "account"])
+  },
+  watch: {}
```
Comment (JonasWijne) [inline src/views/GuestLogin.vue:45]: remove

File: src/views/GuestLogin.vue [hunk 7]
```diff
@@ -0,0 +1,47 @@
+<template>
+  <v-row align="center" justify="center">
+    <v-card :loading="$route.query.callback" width="500">
+      <v-card-title>Freeflow Connect</v-card-title>
+      <v-container>
+        <v-card-text>Choose your guest name</v-card-text>
+        <v-text-field :rules="guestNameRules" v-model="guestName" label="guest" single-line></v-text-field>
+      </v-container>
+      <v-card-actions>
+        <v-btn v-on:click="continueLogin">Continue as Guest</v-btn>
```
Comment (JonasWijne) [inline src/views/GuestLogin.vue:10]: change `v-on:click` to `@click`

File: src/stores/AuthStore.js [hunk 8]
```diff
@@ -42,6 +42,10 @@ export default {
         )}&redirecturl=${encodeURIComponent(redirectUrl)}`
       );
     },
+    async loginAsGuest(context, guestName){
```
[the remark this answers is not in the document]
Comment (jdelrue) [reply src/stores/AuthStore.js:45]: Because I copied it from another function

File: src/views/Login.vue (hunk 5 above)
Comment (jdelrue) [reply src/views/Login.vue:5]: Dont think so

File: src/views/Login.vue [hunk 9]
```diff
@@ -1,55 +1,66 @@
...
-      'loginUrl',
-      'account'
-    ])
+    ...mapGetters(["loginUrl", "account"]),
   },
   watch: {
     loginUrl(val) {
-      if(val){
-        window.location.replace(val)
+      if (val) {
```
Comment (JonasWijne) [inline src/views/Login.vue:51]: flip if

**Comment by JonasWijne** [comment]
:+1:

**Pull request merged**